### ISSUE TITLE
Undefined index - cli - add_tree.php

### ISSUE THREAD

**Issue by gadzet21** [Cacti/cacti#1888]
```
php add_tree.php --list-hosts
Known Devices: (id, hostname, template, description)
PHP Notice:  Undefined index: hostname in /usr/share/cacti/lib/api_automation_tools.php on line 440
PHP Notice:  Undefined index: host_template_id in /usr/share/cacti/lib/api_automation_tools.php on line 440
```
or
```
php add_tree.php --list-sites
PHP Fatal error:  Uncaught Error: Call to undefined function displaySites() in /usr/share/cacti/cli/add_tree.php:149
Stack trace:
#0 {main}
  thrown in /usr/share/cacti/cli/add_tree.php on line 149
```
Cacti 1.2.0 - 2018.09.03

**Comment by cigamit** [comment]
Thanks for testing and reporting.  This resolved now!

**Comment by gadzet21** [comment]
Thanks cigamit.
But I noticed that the add_tree script does not display the full list of hosts (only one IP).

**Comment by gadzet21** [comment]
No errors.
Result from the first query: 98
Result from the second query: 1339

Devices on Cacti: 1248

The script from version 1.1.38 shows all the hosts

**Comment by gadzet21** [comment]
Small correcting - "only one IP" - 7 hosts, but only from one IP

root@cacti2:~# php /usr/share/cacti/cli/add_tree.php --list-hosts
Known Devices: (id, hostname, template, description)
1       xxx.xx.149.10   1       cBR-Olsztyn-1_env_memory
2       xxx.xx.149.10   2       cBR-Olsztyn-1_env_temperature
4       xxx.xx.149.10   4       cBR-Olsztyn-1_traffic_ds
5       xxx.xx.149.10   5       cBR-Olsztyn-1_traffic_us
8       xxx.xx.149.10   8       cBR-Olsztyn-1_env_cpu
9       xxx.xx.149.10   9       cBR-Olsztyn-1_wideband_utilization
10      xxx.xx.149.10   10      cBR-Olsztyn-1_hfcc

**Comment by netniV** [comment]
Yeah I see the problem.  Try correct both queries so that
```sql
ON h.host_template_id = h.id
```
is actually
```sql
ON h.host_template_id = ht.id
```

**Comment by gadzet21** [comment]
First query: 1245
Second query: 1248

**Comment by netniV** [comment]
If you apply the fix in PR #1891 that should fix things up for you.

**Comment by gadzet21** [comment]
Thanks - it's ok now.
